### PR TITLE
cache: added clear_expired_cache (#1055 - #1061)

### PR DESCRIPTION
**Pull request by pl-marasco**
As discussed in #1061 based on the discussion #1055 time to time there is the need to keep data locally and remove them after a defined time.
.clear_expired_cache() add the possibility to remove only the cached data that overpass the defined time.

### CODE REVIEW

File: fsspec/implementations/tests/test_cached.py [hunk 1]
```diff
@@ -153,6 +153,111 @@ def test_clear():
...
+    )
+    assert fs.cat(f4) == data
+
+    cache_fn = os.path.join(fs.storage[-1], "cache")
+    __ager(cache_fn, f4)
+
+    fs.clear_expired_cache()
+    assert not fs._check_file(f4)
+
+    shutil.rmtree(origin)
```
Comment (martindurant) [inline fsspec/implementations/tests/test_cached.py:255]: Maybe use the pytest tmpdir fixture, and subdirectories of that location, to make sure that files are cleaned up even in the case of the test failing.

**Review by martindurant — COMMENTED** [review]
Great, thank you. I just have a couple of small questions.

File: fsspec/implementations/tests/test_cached.py (hunk 1 above)
Comment (pl-marasco) [reply fsspec/implementations/tests/test_cached.py]: updated as requested

File: fsspec/implementations/cached.py [hunk 2]
```diff
@@ -232,6 +232,44 @@ def clear_cache(self):
...
+                    basename = os.path.basename(detail["original"])
+                    fn = os.path.join(self.storage[-1], basename)
+                else:
+                    fn = os.path.join(self.storage[-1], detail["fn"])
+                if os.path.exists(fn):
+                    os.remove(fn)
+                    self.cached_files[-1].pop(path)
+
+        if self.cached_files[-1]:
+            cache_path = os.path.join(self.storage[-1], "cache")
```
[the remark this answers is not in the document]
Comment (pl-marasco) [reply fsspec/implementations/cached.py:266]: I would consider it as a 'simplification' for a specific purpose or in other words, similar but not identical; here we have the self.cache dumped to the file without any comparison with the previous cache file. For my understanding save_cache makes some checks between the file versions (cache) and the object one (self.cached_files) to allow bloks in caching.
Despite all my attempts to modify and reuse .save_cache, I finally found this version to be the more effective solution but I'm open to other opinion and suggestions.

[the remark this answers is not in the document]
Comment (martindurant) [reply fsspec/implementations/cached.py:266]: OK, I'm fine with that - when we explicitly choose to remove old files, we do not need any extra considerations.

**Pull request merged**